### ISSUE TITLE
🐛 Bug Report: Software Templates `dependencies` parameters submitted even if condition not met.

### ISSUE THREAD

**Issue by david-christ** [backstage/backstage#24113]
### 📜 Description

If there are parameter dependencies, i.e. `spec.parameters[].dependencies`, the values entered in the UI to dependent fields get submitted to the scaffolder, even if the condition is not met.

This can happen if there is a dependent field that gets filled out, but then the dependency condition gets un-met afterwards.

### 👍 Expected behavior

When a condition gets un-met when a template gets submitted, the values should be removed.

![Screenshot 2024-04-09 at 13 30 51](https://github.com/backstage/backstage/assets/90661517/8e71f8f5-3b03-4908-b4ac-515e9f8d70a1)

### 👎 Actual Behavior with Screenshots

The values are still present

![Screenshot 2024-04-09 at 13 30 35](https://github.com/backstage/backstage/assets/90661517/3e217ae5-499e-4150-b6f7-983e3cbc9ce2)

### 👟 Reproduction steps

Precondition: a `dependencies` clause, example see context section below

1. Satisfy the condition of the dependency (here, a simple checkbox)
2. Fill out fields from the conditional (here, a description text field)
3. Un-meet the condition (here, untick checkbox)
4. Click "Review"

The dependent fields are still populated, and will also be provided to the scaffolder steps.

![Screenshot 2024-04-09 at 13 30 18](https://github.com/backstage/backstage/assets/90661517/92f1da6a-e24d-42b9-b04c-6c53928dabc3)
![Screenshot 2024-04-09 at 13 30 28](https://github.com/backstage/backstage/assets/90661517/67de1496-a42f-49e0-99f5-34d8daf64e7f)
![Screenshot 2024-04-09 at 13 30 35](https://github.com/backstage/backstage/assets/90661517/3e217ae5-499e-4150-b6f7-983e3cbc9ce2)

### 📃 Provide the context for the Bug.

```yaml
apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: service-test
  title: Test service template
spec:
  owner: platform
  type: service

  parameters:
    - title: Fill in some steps
      required:
        - name
      properties:
        name:
          title: Name
          type: string
          description: Unique name of the service, snake_case
          pattern: '^[a-z_]+$'
          ui:help: 'Example: my_service'
          ui:autofocus: true
        owner:
          title: Owner
          type: string
          description: Team owning the service
          ui:field: OwnerPicker
          ui:options:
            allowArbitraryValues: false
            catalogFilter:
              kind:
                - Group
              spec.type:
                - team
        moreInfo:
          title: Provide description?
          type: boolean
      dependencies:
        # Only show input to generate RPC and worker deployment files when capabilities are supported.
        moreInfo:
          allOf:
            - if:
                properties:
                  moreInfo:
                    const: true
              then:
                properties:
                  description:
                    title: Description
                    type: string
```

### 🖥️ Your Environment

```
OS:   Darwin 23.4.0 - darwin/arm64
node: v18.12.1
yarn: 3.2.3
cli:  0.25.2 (installed)
backstage:  1.23.4

Dependencies:
  @backstage/app-defaults                                          1.5.0
  @backstage/backend-app-api                                       0.5.13, 0.5.14
  @backstage/backend-common                                        0.21.2, 0.21.3
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.6
  @backstage/backend-plugin-api                                    0.6.12, 0.6.13
  @backstage/backend-tasks                                         0.5.17, 0.5.18
  @backstage/backend-test-utils                                    0.3.3
  @backstage/catalog-client                                        1.6.0
  @backstage/catalog-model                                         1.4.4
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.3
  @backstage/cli                                                   0.25.2
  @backstage/config-loader                                         1.6.2
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.12.0
  @backstage/core-compat-api                                       0.2.0
  @backstage/core-components                                       0.14.0
  @backstage/core-plugin-api                                       1.9.0
  @backstage/dev-utils                                             1.0.27
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.5
  @backstage/frontend-plugin-api                                   0.6.0
  @backstage/integration-aws-node                                  0.1.9
  @backstage/integration-react                                     1.1.24
  @backstage/integration                                           1.9.0
  @backstage/plugin-analytics-module-ga4                           0.2.0
  @backstage/plugin-app-backend                                    0.3.61
  @backstage/plugin-app-node                                       0.1.13
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.5
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.4
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.8
  @backstage/plugin-auth-backend-module-github-provider            0.1.10
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.10
  @backstage/plugin-auth-backend-module-google-provider            0.1.10
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.8
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.10
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.6
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.3
  @backstage/plugin-auth-backend-module-okta-provider              0.0.6
  @backstage/plugin-auth-backend                                   0.21.3
  @backstage/plugin-auth-node                                      0.4.7, 0.4.8
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.10
  @backstage/plugin-catalog-backend-module-unprocessed             0.3.10
  @backstage/plugin-catalog-backend                                1.17.3
  @backstage/plugin-catalog-common                                 1.0.21
  @backstage/plugin-catalog-graph                                  0.4.0
  @backstage/plugin-catalog-node                                   1.7.3
  @backstage/plugin-catalog-react                                  1.10.0
  @backstage/plugin-catalog-unprocessed-entities                   0.1.8
  @backstage/plugin-catalog                                        1.17.0
  @backstage/plugin-devtools-backend                               0.2.10
  @backstage/plugin-devtools-common                                0.1.8
  @backstage/plugin-devtools                                       0.1.9
  @backstage/plugin-events-node                                    0.2.22
  @backstage/plugin-github-actions                                 0.6.11
  @backstage/plugin-home-react                                     0.1.8
  @backstage/plugin-home                                           0.6.2
  @backstage/plugin-kubernetes-common                              0.7.4
  @backstage/plugin-org                                            0.6.20
  @backstage/plugin-permission-backend                             0.5.36
  @backstage/plugin-permission-common                              0.7.12
  @backstage/plugin-permission-node                                0.7.23, 0.7.24
  @backstage/plugin-permission-react                               0.4.20
  @backstage/plugin-proxy-backend                                  0.4.11
  @backstage/plugin-scaffolder-backend-module-azure                0.1.5
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.3
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.3
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.3
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.5
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.3
  @backstage/plugin-scaffolder-backend-module-github               0.2.3
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.16
  @backstage/plugin-scaffolder-backend                             1.21.3
  @backstage/plugin-scaffolder-common                              1.5.0
  @backstage/plugin-scaffolder-node                                0.3.3
  @backstage/plugin-scaffolder-react                               1.8.0
  @backstage/plugin-scaffolder                                     1.18.0
  @backstage/plugin-search-backend-module-catalog                  0.1.17
  @backstage/plugin-search-backend-module-pg                       0.5.22
  @backstage/plugin-search-backend-node                            1.2.17
  @backstage/plugin-search-backend                                 1.5.3
  @backstage/plugin-search-common                                  1.2.10
  @backstage/plugin-search-react                                   1.7.6
  @backstage/plugin-search                                         1.4.6
  @backstage/plugin-user-settings                                  0.8.1
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.6.3
  @backstage/test-utils                                            1.5.0
  @backstage/theme                                                 0.5.1
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by benjdlambert** [comment]
Ah I think that this is because we currently don't use the `omitExtraData` and `liveOmit` props to the form. 

Do you want to raise a PR to these props to the `Stepper` component?

**Comment by benjdlambert** [comment]
Fixed by #24826
